web/src/components/Settings/MemberSection.tsx

```diff
@@ -24,29 +24,24 @@ const PreferencesSection = () => {
   });
   const [userNameQueryText, setUserNameQueryText] = useState("");
   const [userList, setUserList] = useState<User[]>([]);
-  const [showUserList, setShowUserList] = useState<User[]>([]);
+
+  const showUserList = userList.filter((user: User) => {
+    return user.username.toLowerCase().includes(userNameQueryText.toLowerCase());
+  });
 
   useEffect(() => {
     fetchUserList();
-    setShowUserList(userList);
   }, []);
 
   useEffect(() => {
-    filterUserList();
-  }, [userNameQueryText, userList]);
+    fetchUserList();
+  }, [userNameQueryText]);
```

Comment:
We fetch all member in one request. So when query text changed, we don't need to fetch data again.